improver/nbhood/use_nbhood.py

```diff
@@ -402,8 +403,10 @@ def process(self, cube):
         # Loop over any extra dimensions
         cubelist = iris.cube.CubeList([])
         for slice_3d in cube.slices([self.coord_masked, yname, xname]):
-            collapsed_slice = slice_3d.collapsed(
-                self.coord_masked, iris.analysis.MEAN, weights=weights)
+
```

Comment:
Such a cold lonely void. Why must I be so far from my parent for loop? Please reunite us.

Reply:
Back with family. I'm sorry `code`, please forgive me.